lib/scala/connected-lock-manager/src/test/scala/org/enso/lockmanager/ConnectedLockManagerTest.scala

```diff
@@ -150,11 +150,31 @@ class ConnectedLockManagerTest
       }
 
       sync.join()
-      sync.summarizeReports() shouldEqual Seq(
-        "primary-acquired",
-        "connected-acquired",
-        "released",
-        "released"
+      sync.summarizeReports() should (
+        equal(
+          Seq(
+            "primary-acquired",
+            "connected-acquired",
+            "released",
+            "released"
+          )
+        ) or
+        equal(
+          Seq(
+            "connected-acquired",
+            "primary-acquired",
+            "released",
+            "released"
+          )
+        ) or
+        equal(
+          Seq(
+            "connected-acquired",
+            "released",
+            "primary-acquired",
+            "released"
+          )
+        )
```

Comment:
These 2 scenarios should not be possible at all, given that `report("connected-acquired")` is run _after_ `waitFor("primary-acquired")`. The whole purpose of that `waitFor` is to ensure expected ordering of the operations regardless of thread scheduling, to ensure that the test does indeed find both locks in the critical section at the same time.

For the same reason, the 'primary' will not release before `connected-acquired` is reported - thus the scenario where the release happens _before_ `connected-acquired` should not happen either.

Reply:
That's not correct. In the current setup they are very much possible.